### ISSUE TITLE
Possible to get the parameter settings to reproduce all of the reported results in the paper?

### ISSUE THREAD

**Issue by nlml** [haeusser/learning_by_association#6]
Hi there,

First off - great paper and idea. I think it's a really creative use of neural nets for semi-supervised learning.

I'm looking into possible extensions of your work. To do so it would be good if I could pivot off the results reported in your paper as baselines. I could go through and try to figure out all the parameter settings I need to use myself. Instead though, it would be a lot easier if I already had the exact parameter settings.

Could you possibly post all of the parameter arguments that I should provide to your code to reproduce all of the results in the paper? Or at least, those coming close to or beating state of the art on SVHN (particularly 500/1k labels) and on MNIST (particularly 100 labels, permutation invariant and convolutional)? It would be really helpful.

Kind regards,
Liam Schoneveld

**Comment by haeusser** [comment]
Hi @nlml 

Thank you for reaching out and suggesting a hyper param overview. I should write some scripts with the respective hyper params some time in the next few weeks. For the particular SVHN experiment you mentioned, I had the following setup:

- visit_weight = 0.25
- augmentation = True
- sup_per_class = [2, 10, 50, 100, 200, -1]
- unsup_samples = [0, 1000, 20000, -1]

The learning rates were:

- Beginning: 1e-3
- After 1500 steps: 1e-4
- After 100000s steps: 1e-5

The other hyper params correspond to the flag defaults.

I hope this helps. Cheers,
Philip

**Comment by nlml** [comment]
Thanks a lot Philip - and yes that would be great if you could publish the remaining hyperparameters when you get a chance.

Cheers,
Liam

**Comment by haeusser** [comment]
I used `tf.train.piecewise_constant`. By the time I pushed the code, there was a bug with the open source version of TensorFlow. But it should be easy to implement it now. 

Just looked at my logs, I let it run for 700k steps but I think it converged way before then ;-)

**Comment by nlml** [comment]
Thanks :)

So I've run this with these params:

python semisup/train.py \
 --augmentation=True \
 --sup_per_class=50	 \
 --unsup_samples=-1 \
 --visit_weight=0.25 \
 --custom_svhn_lr=True \
 --max_steps=400000 \
 --logdir=./log/svhn-best-50perclass

Where I also have:

if not FLAGS.custom_svhn_lr:
  t_learning_rate = tf.maximum(
      tf.train.exponential_decay(
          FLAGS.learning_rate,
          model.step,
          FLAGS.decay_steps,
          FLAGS.decay_factor,
          staircase=True),
      FLAGS.minimum_learning_rate)

And after 100k steps my loss is at 3.5 (and may still be slowly going down), while my test set accuracy is only around 87.4%, whereas the result in the paper is more like 93.75% (6.25% error).

Does this sound right to you? Or am I doing something wrong perhaps?

**Comment by yfllllll** [comment]
@nlml I have the same problem with you. Have you reproduced the results?

**Comment by nlml** [comment]
Nope, never managed to reproduce svhn results in paper

**Comment by haeusser** [comment]
You have to implement your own piecewise learning rate schedule according to the description in the paper. The code was written during my internship at Google with a non-public version of TensorFlow. I wish I could have published the exact code. Sorry about this.

**Comment by yfllllll** [comment]
Hi @haeusser, I run the code for semi-supervised learning on svhn with/without data augmentation. And my test set accuracy are 90+%, 89+%, respectively. There maybe a problem with my data augmentation code. Can you share the data augmentation code here?
